src/finance/MainnetSwapSteward.sol

```diff
@@ -0,0 +1,145 @@
+ * @author efecarranza  (Tokenlogic)
+ * @notice Facilitates token swaps on behalf of the DAO Treasury using the AaveSwapper.
+ * Funds must be present in the AaveSwapper in order for them to be executed.
+ * The tokens that are to be swapped from/to are to be pre-approved via governance.
+ *
+ * -- Security Considerations
+ * Having previously set and validated oracles avoids mistakes that are easy to make when passing the necessary parameters to swap.
+ *
+ * -- Permissions
+ * The contract implements OwnableWithGuardian.
+ * The owner will always be the respective network Short Executor (governance).
+ * The guardian role will be given to a Financial Service provider of the DAO.
+ *
+ */
+contract MainnetSwapSteward is OwnableWithGuardian, IMainnetSwapSteward {
+  using CU for CU.SwapInput;
+
+  /// @inheritdoc IMainnetSwapSteward
+  uint256 public constant MAX_SLIPPAGE = 1000; // 10%
+
+  /// @inheritdoc IMainnetSwapSteward
+  AaveSwapper public immutable SWAPPER;
+
+  /// @inheritdoc IMainnetSwapSteward
+  ICollector public immutable COLLECTOR;
+
+  /// @inheritdoc IMainnetSwapSteward
+  address public MILKMAN;
+
+  /// @inheritdoc IMainnetSwapSteward
+  address public PRICE_CHECKER;
+
+  /// @inheritdoc IMainnetSwapSteward
+  mapping(address token => bool isApproved) public swapApprovedToken;
+
+  /// @inheritdoc IMainnetSwapSteward
+  mapping(address token => address oracle) public priceOracle;
+
+  constructor(address _owner, address _guardian, address collector) {
+    _transferOwnership(_owner);
+    _updateGuardian(_guardian);
+
+    COLLECTOR = ICollector(collector);
+    SWAPPER = AaveSwapper(MiscEthereum.AAVE_SWAPPER);
+
+    // https://etherscan.io/address/0x060373D064d0168931dE2AB8DDA7410923d06E88
+    _setMilkman(0x060373D064d0168931dE2AB8DDA7410923d06E88);
+
+    // https://etherscan.io/address/0xe80a1C615F75AFF7Ed8F08c9F21f9d00982D666c
+    _setPriceChecker(0xe80a1C615F75AFF7Ed8F08c9F21f9d00982D666c);
+  }
+
+  /// @inheritdoc IMainnetSwapSteward
+  function tokenSwap(address sellToken, uint256 amount, address buyToken, uint256 slippage)
+    external
+    onlyOwnerOrGuardian
+  {
+    _validateSwap(sellToken, amount, buyToken, slippage);
+
+    CU.SwapInput memory swapData = CU.SwapInput(
+      MILKMAN, PRICE_CHECKER, sellToken, buyToken, priceOracle[sellToken], priceOracle[buyToken], amount, slippage
+    );
+
+    CU.swap(COLLECTOR, address(SWAPPER), swapData);
+  }
+
+  /// @inheritdoc IMainnetSwapSteward
+  function setSwappableToken(address token, address priceFeedUSD) external onlyOwner {
+    if (priceFeedUSD == address(0)) revert MissingPriceFeed();
+
+    swapApprovedToken[token] = true;
+    priceOracle[token] = priceFeedUSD;
+
+    // Validate oracle has necessary functions
+    if (IAggregatorInterface(priceFeedUSD).decimals() != 8) {
+      revert PriceFeedIncompatibility();
+    }
+    if (IAggregatorInterface(priceFeedUSD).latestAnswer() == 0) {
+      revert PriceFeedIncompatibility();
```

Comment:
it's a bad practice to return the same error on different places of the same function.
Any auditor would recommend to either combine the conditional or to return explicit errors.
I'd recommend to go with the latter `PriceFeedIncompatibleDecimals` and `PriceFeedInvalidAnswer` or similar.